var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -184,6 +190,22 @@ def configure_args(self):
             ])
             if '@:2.20.99' in spec:
                 configure_args.append('--disable-cub')
+        if '+rocm' in spec:
+            configure_args.extend([
+                '--with-hip',
+                '--enable-rocrand',
+                '--enable-rocsparse'
```

Comment:
I would think you might need:

```
    depends_on('hip', when='+rocm')
    depends_on('rocrand', when='+rocm')
    depends_on('rocsparse', when='+rocm')
```

And if  spack were to install these packages - their location won't be the same as `hip`  so somehow the appropriate includes/libs need to be passed in.

```[balay@instinct spack]$ spack find -p hip rocrand rocsparse
==> 3 installed packages
-- linux-centos7-broadwell / gcc@10.2.0 -------------------------
hip@4.3.1        /storage/users/balay/spack/opt/spack/linux-centos7-broadwell/gcc-10.2.0/hip-4.3.1-2skkbesaligdhmc2bx4p4cszjnukueyc
rocrand@4.3.1    /storage/users/balay/spack/opt/spack/linux-centos7-broadwell/gcc-10.2.0/rocrand-4.3.1-cp64x43alpb6vomimopg2z3foitncwul
rocsparse@4.3.1  /storage/users/balay/spack/opt/spack/linux-centos7-broadwell/gcc-10.2.0/rocsparse-4.3.1-uo2ptfp6akukt3kvfumqt352tmeq4aet
```
My changes (for these issues) for PETSc are at https://github.com/spack/spack/pull/27779

Reply:
Thanks @balay! Currently, we're assuming `rocrand` and `rocsparse` are located where `hip` is installed, so we'd need to make the change to allow this within hypre.